be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala

```diff
@@ -39,12 +39,12 @@ import java.io.InputStream
 object Validator {
 
   private final val queryPath = "protocol/query/query.json" //With respect to resource folder
-  final val objectMapper: ObjectMapper = new ObjectMapper()
-  final lazy val schema: JsonSchema = setupSchemaValidation
+  private final val objectMapper: ObjectMapper = new ObjectMapper()
+  private final lazy val schema: JsonSchema = setupSchemaValidation(queryPath, objectMapper)
 
-  def setupSchemaValidation : JsonSchema = {
+  def setupSchemaValidation(jsonPath: String, objectMapper: ObjectMapper): JsonSchema = {
```

Comment:
question, do you use the objectMapper argument somewhere?

Reply:
Yes it's used in line 53. (You'll have to expand the file to see it)

Reply:
oh right